src/indexer/segment_updater.rs

```diff
@@ -468,21 +456,20 @@ impl SegmentUpdater {
         files
     }
 
-    pub(crate) async fn schedule_commit(
+    pub(crate) fn schedule_commit(
         &self,
         opstamp: Opstamp,
         payload: Option<String>,
-    ) -> crate::Result<()> {
+    ) -> FutureResult<Opstamp> {
         let segment_updater: SegmentUpdater = self.clone();
-        self.schedule_task(async move {
+        self.schedule_task(move || {
             let segment_entries = segment_updater.purge_deletes(opstamp)?;
             segment_updater.segment_manager.commit(segment_entries);
             segment_updater.save_metas(opstamp, payload)?;
-            let _ = garbage_collect_files(segment_updater.clone()).await;
-            segment_updater.consider_merge_options().await;
-            Ok(())
+            let _ = garbage_collect_files(segment_updater.clone());
+            segment_updater.consider_merge_options();
```

Comment:
previously they were executed serially, now it's fire and forget

Reply:
not really. previously the function were async for shitty reasons,. now they are blocking.

Reply:
that's quite confusing, shouldn't there be a `wait()`?

Reply:
Ah yes the first one needs to be wait !

Reply:
ah no both are blocking. There is no issue here.